cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -47,15 +47,17 @@ func runProfile(cmd *cobra.Command, args []string) {
 			atexit.ExitWithMessage(1, err.Error())
 		}
 	} else {
-		err := cmdUtil.SetOcContext(profileName)
-		if err != nil {
-			if glog.V(2) {
-				fmt.Println(fmt.Sprintf("%s", err.Error()))
+		if cmdUtil.GetVMStatus(profileName) == "Running" {
```

Comment:
As I mentioned before, I think we also need to unset the currently active context from the previous profile (if set). 

If I switch to another profile, the current oc CLI context needs to change. If possible it should switch to a new context, but in the case where the profile does not exist or the profile VM is not running, the current CLI context needs to be unset. Otherwise one rins the risk that the user runs oc commands which are intended for a new conext against the old context.

Reply:
@hferentschik i couldn't find any helper function to unset oc cli context, the context will be switched upon `minishift start`.

Reply:
@hferentschik we would also need a way to find out the current context then.

Reply:
>  i couldn't find any helper function to unset oc cli context, the context will be switched upon minishift start.

Most likely needs to be written. 

> we would also need a way to find out the current context then.

True. We might have to. All this can be done though. How much of the required infrastructure is already in place is a different question.

Reply:
I need to understand first what the setting/unsetting cli context actually do in the code, i'll look into it and reach out to you for further questions.

Reply:
@hferentschik there is `oc config current-context` commnad to get the current context, however i don't know how to unset the context as the context cannot be empty AFAIU.